mesonbuild/mcompile.py

```diff
@@ -27,9 +29,45 @@
 if T.TYPE_CHECKING:
     import argparse
 
+def generate_ninja_target_name(target_name: str, builddir: Path, introspect_data: dict) -> str:
+    for target in introspect_data:
+        if (not target['subproject'] and target['name'] == target_name):
+            target_type = target['type']
+            if (target_type == 'run'):
+                return target_name
+            elif (target_type == 'custom'):
+                # Custom targets do not have a clear mapping to `ninja` targets
+                raise MesonException('Can\'t invoke target `{}`: target of type `{}` can\'t be invoked directly on this backend'.format(target_name, target_type))
+            else:
+                if (target['filename'] == 'no_name'):
+                    raise MesonException('Can\'t invoke target `{}`: unexpected target configuration: missing filename'.format(target_name))
+                 
+                return str(Path(target['filename'][0]).relative_to(builddir.resolve()))
```

Comment:
shouldn't this be in the form:
```python
elif target['filename'] == 'no_name':
     raise ...
else:
     return ...
```

Reply:
Well, I prefer making a clear distinction between the purpose of each if\else group.
E.g. I would've used `elif` here if it was handling the same condition parameter like this:
```
if target.name == 'a':
  ...
elif target.name == 'b':
  ...
```
I can change it if you want though, since it's just a style choice.

Reply:
I guess either way then